src/urllib3/contrib/pyopenssl.py

```diff
@@ -61,23 +63,48 @@ class UnsupportedExtension(Exception):  # type: ignore[no-redef]
 from .. import util
 
 if typing.TYPE_CHECKING:
-    from OpenSSL.crypto import X509  # type: ignore[import]
+    from OpenSSL.crypto import X509  # type: ignore[import-untyped]
 
 
 __all__ = ["inject_into_urllib3", "extract_from_urllib3"]
 
+
+class SSLMethodEnum(Enum):
+    PROTOCOL_TLS = util.ssl_.PROTOCOL_TLS  # type: ignore[attr-defined]
+    PROTOCOL_TLS_CLIENT = util.ssl_.PROTOCOL_TLS_CLIENT  # type: ignore[attr-defined]
+    PROTOCOL_TLSv1 = ssl.PROTOCOL_TLSv1
+    PROTOCOL_TLSv1_1 = (
+        ssl.PROTOCOL_TLSv1_1 if hasattr(ssl, "PROTOCOL_TLSv1_1") else None
+    )
+    PROTOCOL_TLSv1_2 = (
+        ssl.PROTOCOL_TLSv1_2 if hasattr(ssl, "PROTOCOL_TLSv1_2") else None
+    )
+
+
 # Map from urllib3 to PyOpenSSL compatible parameter-values.
 _openssl_versions = {
```

Comment:
Can this be an alternative to the extensive changes to the file?

```suggestion
_openssl_versions: dict[int, int] = {
```